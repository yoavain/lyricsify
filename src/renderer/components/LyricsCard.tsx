import * as React from "react";
import { Card, CardContent, Typography } from "@material-ui/core";

interface LyricsCardProps {
    title?: string;
    artist?: string;
    lyrics?: string;
}

const prepareLyrics = (lyrics: string | undefined) => {
    if (!lyrics) {
        return <div>[Missing Lyrics]</div>;
    }

    return lyrics.split("\n").map((line, index) => <div key={index}>{line}</div>);
};

const LyricsCard = (props: LyricsCardProps) => {
    console.log(`Rendering LyricsCard ${JSON.stringify({ ...props, lyrics: !!props.lyrics })}`);
    return (
        <Card>
            <CardContent>
                <Typography variant="h4" component="h2">
                    {props.title} - Lyrics
                </Typography>
                <Typography variant="h5" component="h2">
                    {props.artist}
                </Typography>
                <Typography variant="h6" component="h2">
                    {prepareLyrics(props.lyrics)}
                </Typography>
            </CardContent>
        </Card>
    );
};

export default LyricsCard;