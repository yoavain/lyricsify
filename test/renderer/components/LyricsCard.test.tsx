import * as React from "react";
import LyricsCard from "~components/LyricsCard";
import { render } from "@testing-library/react";

describe("LyricsCard component", () => {
    it("renders correctly - no lyrics", () => {
        const root = render(<LyricsCard />);
        expect(root).toMatchSnapshot();
    });
    it("renders correctly - with lyrics", () => {
        const root = render(<LyricsCard title="Title" artist="Artist" lyrics="These are the lyrics"/>);
        expect(root).toMatchSnapshot();
    });
});
