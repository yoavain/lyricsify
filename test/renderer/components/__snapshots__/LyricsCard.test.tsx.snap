// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LyricsCard component renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h4"
    >
       - Lyrics
    </h2>
    <h2
      className="MuiTypography-root MuiTypography-h5"
    />
    <h2
      className="MuiTypography-root MuiTypography-h6"
    >
      <div>
        [Missing Lyrics]
      </div>
    </h2>
  </div>
</div>
`;

exports[`LyricsCard component renders correctly 2`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h4"
    >
      Title
       - Lyrics
    </h2>
    <h2
      className="MuiTypography-root MuiTypography-h5"
    >
      Artist
    </h2>
    <h2
      className="MuiTypography-root MuiTypography-h6"
    >
      <div>
        These are the lyrics
      </div>
    </h2>
  </div>
</div>
`;
